gmsh_utils/gmsh_IO.py

```diff
@@ -154,103 +155,182 @@ def create_surface(self, line_ids: Union[List[int], npt.NDArray[np.int_]], name_
+        list_lines = []
+        for point_pair in range(len(point_pairs)):
+            line = [point_pairs[point_pair][0], point_pairs[point_pair][1]]
+            line_id = self.create_line(line)
+            list_lines.append(line_id)
+        return list_lines
+
+    def make_surfaces(self, line_list: Union[List[int], npt.NDArray[np.int_]], name_label: str) \
+            -> Union[List[int], npt.NDArray[np.int_]]:
         """
-        Creates 3D geometries by extruding the 2D surface
+        Makes surfaces with surface tags by getting line tags.
 
         Args:
-            point_coordinates (Union[List[float], npt.NDArray[float]]): Geometry points coordinates.
-            point_pairs (Union[List[List[int]], npt.NDArray[npt.NDArray[int]]]): A list of point tags of two consecutive
-                points in an array.
-            mesh_size (float): Mesh size.
-            extrusion_length (Union[List[float], npt.NDArray[float]]): The extrusion length in x, y and z direction.
-            name_label (str): surface name label from user input.
+            line_list (Union[List[int], npt.NDArray[np.int_]]): A list of line tags in order.
+            name_label (str): surface name labels provided by user input.
+
+        Returns:
+            List[int]: A list of surface tags.
+        """
+
+        surfaces = []
+        surfaces.append(self.create_surface(line_list, name_label))
+
+        return surfaces
+
+    def make_volume(self, surface_id: Union[List[int], npt.NDArray[np.int_]],
+                    extrusion_length: Union[List[float], npt.NDArray[np.float64]], name_label: str) -> None:
+        """
+        Makes volume with volume tags by getting surface tags.
+
+        Args:
+            surface_id (Union[List[int], npt.NDArray[np.int_]]): A list of surface tags.
+            extrusion_length (Union[List[float], npt.NDArray[np.float64]]): The extrusion length in x, y and z direction
+            name_label (str): A list of volume name labels provided by user input.
 
         Returns:
             None
         """
 
-        surfaces = self.make_geometry_2D(point_coordinates, point_pairs, mesh_size, name_label)
-        self.create_volume_by_extruding_surface(surfaces, extrusion_length)
+        volumes = []
+        for volume in range(len(surface_id)):
+            volume_tag = self.create_volume_by_extruding_surface(surface_id[volume], extrusion_length,
+                                                                 name_label[volume], volume)
+            volumes.append(volume_tag)
 
-    def generate_point_pairs(self, point_coordinates: Union[List[List[float]], npt.NDArray[np.float64]]) \
-            -> List[List[int]]:
+    def make_geometry_2d(self, point_coordinates: Union[List[float], Any],
+                         default_mesh_size: float, name_label_list: str) -> Union[List[int], npt.NDArray[np.int_]]:
```

Comment:
can you make default_mesh_size optional. For this you can set the default as: default_mesh_size: float = -1.

I think gmsh checks if the mesh size >0, if not gmsh chooses the mesh size itself, based on what it thinks is logical

Same for all the other places where default_mesh_size is called